beancount_interpolate/common.py

```diff
@@ -118,7 +123,7 @@ def parse_mark(mark, default_date, config):
 
 # Infer Duration, start and steps. Spacing optinonal. Format: [123|KEYWORD] [@ YYYY-MM[-DD]] [/ step]
 # 0. max duration, 1. year, 2. month, 3. day, 4. min step
-RE_PARSING = re.compile(r"^\s*?([^-/\s]+)?\s*?(?:@\s*?([0-9]{4})-([0-9]{2})(?:-([0-9]{2}))?)?\s*?(?:\/\s*?([^-/\s]+)?\s*?)?$")
+RE_PARSING = re.compile(r"^\s*([0-9]+(?=[\sa-zA-Z]+)(?!\s*[@$/]))?\s*?([^-/\s]+)?\s*?(?:@\s*?([0-9]{4})-([0-9]{2})(?:-([0-9]{2}))?)?\s*?(?:\/\s*?([0-9]+)?\s*([^-/\s]+)?\s*?)?$")
```

Comment:
This regex was afwul to start with, but now it's beyond understandable :D Can you please document it? For example like this 
```
E_PARSING = re.compile(r"^\s*([0-9]+(?=[\sa-zA-Z]+)(?!\s*[@$/]))?\s*?([^-/\s]+)?\s*?(?:@\s*?([0-9]{4})-([0-9]{2})(?:-([0-9]{2}))?)?\s*?(?:\/\s*?([0-9]+)?\s*([^-/\s]+)?\s*?)?$")
#                            ( 1.                              )     ( 2.     )
# 1. the first thing doing it's thing
# 2. the second thing matching another thing
# 3. ...
```

Reply:
Yes I can do that. I would then like to split the regex into multiple lines surrounded by braces.

Reply:
This is then way easier to annotate

Reply:
I also noticed that are some unnecessary `?` and maybe some other stuff.

Reply:
A small test suite for different meta tags would be indeed quite nice.